sql: show indpred for partial indexes on pg_catalog

otan opened the pull request:
Release note (sql change): Show indpred on the pg_index table for
partial indexes. This was previously NULL, even for partial indexes.

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70808#-:-MkhVJhDDe_KEirA_t7q:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/sql/pg_catalog.go, line 1733 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70808#-MkhV2gZ7ao3S7JaLvUk:-MkhV2gZ7ao3S7JaLvUl:b-j1sxju) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b6ce0433ec9b9c4401a7b262414b9a689b105e9/pkg/sql/pg_catalog.go#L1733)):*
> ```Go
> 					}
> 					indpred := tree.DNull
> 					if pred := index.GetPredicate(); pred != "" {
> ```

nit: you can use `index.IsPartial()` which might read a bit easier, then do `index.GetPredicate()` below.


<!-- Sent from Reviewable.io -->

otan committed:
sql: show indpred for partial indexes on pg_catalog

Release note (sql change): Show indpred on the pg_index table for
partial indexes. This was previously NULL for partial indexes.

otan commented:
bors r=rafiss,mgartner
